app/views/users/index.html.erb

```diff
@@ -1,5 +1,16 @@
 <h1>User page = user-> index.html</h1>
 
+<div class="nav">
+    <%if admin_signed_in? %>
+      <%= button_to "Sign Out", destroy_admin_session_path, method: :delete %>
+
+     <button> <%= link_to "Edit your profile", edit_admin_registration_path %></button>
+
+    <% else %>
+      <%= link_to "Sign In",  new_admin_session_path %>
+    <% end %>
+</div>
+
 <div>
      <%@user.each do |person|%>
         <%= link_to user_path(person.id) do%>
```

Comment:
The `users/index` displays nothing although I created two users.

![Screenshot from 2022-11-12 13-33-23](https://user-images.githubusercontent.com/38088818/201472144-a7e6d5d2-5a92-483d-bd7b-b95175c27dfa.png)

The are a couple of reasons why that is from the following analysis:
- You've created a new table `admins` which now has all the information for newly registered users, password confirmations, and the column `role`. This is all unnecessary since all this information should be in the `users` table.

![Screenshot from 2022-11-12 13-40-03](https://user-images.githubusercontent.com/38088818/201472280-b45dd364-e3b1-4cf3-81e3-d4d3196e6c6a.png)

As a result the `users` table no longer has a purpose since it always remains empty since all the new users are saved to the `admins` table. And since the `users_controller.rb` fetches data from the `users` table, it always returns an empty array.

![Screenshot from 2022-11-12 13-42-13](https://user-images.githubusercontent.com/38088818/201472364-a164c6c7-8b5d-4ae6-b072-a05a6ac834de.png)

- I think you've approached this the wrong way, and it complicates everything for no reason. I would suggest that you keep all registered users in the `users` table, the way everything was before. You don't really need to manage admins since you will not be creating admin users for now.

Your setup can be as simple as keeping all registration in the `users` table:

![Screenshot from 2022-11-12 13-54-18](https://user-images.githubusercontent.com/38088818/201472781-7aabab4e-7935-45aa-b7f8-4e3c39674592.png)

Having the ability class in `app/models/ability.rb` look like this:
```
 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  //owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end
```
OR
```
....
      can :manage, Comment, author_id: user.id
      can :manage, Post, author_id: user.id
...
```

Then create the `admin` method inside the user model

```
class User < ApplicationRecord
   ...
  def admin?
     role == 'admin' ? true : false
   end
end
```

Reply:
Than you very much but everything is working well on my side. My devise model is called 'admin' and not 'user'.
 Below are screen shots :

![Screenshot (108)](https://user-images.githubusercontent.com/63410324/201480982-7e269e06-e4c1-4ad3-8171-2b454a5e168e.png)

![Screenshot (109)](https://user-images.githubusercontent.com/63410324/201480998-c4f2c34e-1803-4b6c-ab79-0f195d02c486.png)

![Screenshot (110)](https://user-images.githubusercontent.com/63410324/201481020-6cac03fb-6896-4d0f-89fd-e8e210a0c609.png)